crates/ruff_python_formatter/tests/snapshots/format@statement__list_comp.py.snap

```diff
@@ -0,0 +1,72 @@
+---
+source: crates/ruff_python_formatter/tests/fixtures.rs
+input_file: crates/ruff_python_formatter/resources/test/fixtures/ruff/statement/list_comp.py
+---
+## Input
+```py
+[i for i in []]
+
+[i for i in [1,]]
+
+[
+    a  # a
+    for # for
+    c  # c
+    in  # in
+    e  # e
+]
+
+[
+    # before a
+    a  # a
+    # before for
+    for # for
+    # before c
+    c  # c
+    # before in
+    in  # in
+    # before e
+    e
+    # before end
+]
+```
+
+## Output
+```py
+[i for i in []]
+
+[
+    i
+    for
+    i
+    in
+    [
+        1,
+    ]
+]
```

Comment:
black formats this as 
```python
[
    i
    for i in [
        1,
    ]
]
```
which i prefer, but i expect that to be unreasonably hard to implement (breaking the inner group but not the outer group)

Reply:
Hmm interesting. I think this goes into the one case that my expression formatting refactor doesn't cover yet (and I also don't fully understand yet). 

Black sometimes (for example on expression statement level) only expands by parenthesized expressions, but not by operators. But it does sometimes and inserts optional parentheses. I haven't understood the rules around this yet.  It could be related and we may want to wait with addressing it just yet